x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java

```diff
@@ -27,7 +27,8 @@ public Set<NodeFeature> getFeatures() {
             TextSimilarityRankRetrieverBuilder.TEXT_SIMILARITY_RERANKER_RETRIEVER_SUPPORTED,
             RandomRankRetrieverBuilder.RANDOM_RERANKER_RETRIEVER_SUPPORTED,
             SemanticTextFieldMapper.SEMANTIC_TEXT_SEARCH_INFERENCE_ID,
-            SemanticQueryBuilder.SEMANTIC_TEXT_INNER_HITS
+            SemanticQueryBuilder.SEMANTIC_TEXT_INNER_HITS,
+            SemanticTextFieldMapper.SEMANTIC_TEXT_DEFAULT_ELSER_2
```

Comment:
Based on the naming of this feature, do we have a plan on how we will handle defaulting to ELSER future versions of ELSER are released? 

Here's why I ask this: I suspect that we'll want the workflow to be, "Update the default for all new indices created with semantic text to use ELSERvNEXT, and existing indices where inference has already happened can and will continue to use their configured inference ID." Which sounds right to me. But if that's our plan, we can probably simplify this code for the future by removing the `_2` everywhere and simply saying that we're defaulting to ELSER. WDYT?

Reply:
Yes, we have a plan for how we will update to default ELSER 3:

- Add a new default inference endpoint for ELSER 3 (probably `.elser-3`)
- Update semantic text to use ELSER 3 by default for all new indices. Per your point, all pre-existing indices will continue to use ELSER 2 by default.

Since we are currently defaulting to ELSER 2, we want that information communicated in all of the relevant names used (inference endpoints, cluster features, etc). When we update to default ELSER 3, we will do the same (i.e. cluster feature will be `semantic_text.default_elser_3`). If we were to remove the `2` suffix, we would jump from no suffix to `3`, which could add a lot of confusion.

Reply:
Got it, so we plan to communicate it more through this. That makes sense, thanks for explaining!